lib/stat_tracker.rb

```diff
@@ -121,7 +126,22 @@ def best_offense
   end
 
   def worst_offense
-    top_team_array = offense_helper.min_by {|team, score| score}
+    worst_team_array = offense_helper.min_by {|team, score| score}
+    worst_team_id = worst_team_array[0].to_s
+    worst_team = @teams.find {|team| team.id == worst_team_id }
+    worst_team.team_name
+  end
+
+  def highest_scoring_visitor
+    @grouped_game_teams
+    scores_when_visitor = {}
+    @grouped_game_teams.each do |team, games|
```

Comment:
This one was a doozy. The choice break this up into helper methods was a good decision.